lib/pos_blockchain.go

```diff
@@ -12,8 +12,29 @@ import (
 	"github.com/pkg/errors"
 )
 
+// ProcessHeaderPoS simply acquires the chain lock and calls processHeaderPoS.
+func (bc *Blockchain) ProcessHeaderPoS(header *MsgDeSoHeader) (_isMainChain bool, _isOrphan bool, _err error) {
+	// Grab the chain lock
+	bc.ChainLock.Lock()
+	defer bc.ChainLock.Unlock()
+
+	if header == nil {
+		return false, false, fmt.Errorf("ProcessHeaderPoS: Header is nil")
+	}
+
+	headerHash, err := header.Hash()
```

Comment:
Why not push the header hashing into `processHeaderPoS`? This way you don't have to worry about the caller passing through any funny business.